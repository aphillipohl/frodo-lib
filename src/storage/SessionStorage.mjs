/* eslint-disable no-return-assign */
import fs from 'fs';
// import pkg from '../../package.json' assert { type: 'json' };

const pkg = JSON.parse(
  fs.readFileSync(new URL('../../package.json', import.meta.url))
);

const _sessionStorage = {};

export default {
  session: {
    setItem: (key, value) => (_sessionStorage[key] = value),
    getItem: (key) => _sessionStorage[key],
    removeItem: (key) => delete _sessionStorage[key],
    raw: _sessionStorage,
    setUsername: (value) => (_sessionStorage.username = value),
    getUsername: () => _sessionStorage.username,
    setPassword: (value) => (_sessionStorage.password = value),
    getPassword: () => _sessionStorage.password,
    setTenant: (value) => (_sessionStorage.tenant = value),
    getTenant: () => _sessionStorage.tenant,
    setDeploymentType: (value) => (_sessionStorage.deploymentType = value),
    getDeploymentType: () => _sessionStorage.deploymentType,
    setRealm: (value) => (_sessionStorage.realm = value),
    getRealm: () => _sessionStorage.realm,
    setCookieName: (value) => (_sessionStorage.cookieName = value),
    getCookieName: () => _sessionStorage.cookieName,
    setCookieValue: (value) => (_sessionStorage.cookieValue = value),
    getCookieValue: () => _sessionStorage.cookieValue,
    setBearerToken: (value) => (_sessionStorage.bearerToken = value),
    getBearerToken: () => _sessionStorage.bearerToken,
    setLogApiKey: (value) => (_sessionStorage.logApiKey = value),
    getLogApiKey: () => _sessionStorage.logApiKey,
    setLogApiSecret: (value) => (_sessionStorage.logApiSecret = value),
    getLogApiSecret: () => _sessionStorage.logApiSecret,
    setAmVersion: (value) => (_sessionStorage.amVersion = value),
    getAmVersion: () => _sessionStorage.amVersion,
    setFrodoVersion: (value) => (_sessionStorage.frodoVersion = value),
    getFrodoVersion: () =>
      _sessionStorage.frodoVersion || `v${pkg.version} [${process.version}]`,
    setAllowInsecureConnection: (value) => (_sessionStorage.insecure = value),
    getAllowInsecureConnection: () => _sessionStorage.insecure,
    setConnectionProfilesPath: (value) =>
      (_sessionStorage.connectionProfilesPath = value),
    getConnectionProfilesPath: () => _sessionStorage.connectionProfilesPath,
    setMasterKeyPath: (value) => (_sessionStorage.masterKeyPath = value),
    getMasterKeyPath: () => _sessionStorage.masterKeyPath,
    setPrintHandler: (printHandler) =>
      (_sessionStorage.printHandler = printHandler),
    getPrintHandler: () => _sessionStorage.printHandler,
    setCreateProgressHandler: (handler) =>
      (_sessionStorage.createProgressHandler = handler),
    getCreateProgressHandler: () => _sessionStorage.createProgressHandler,
    setUpdateProgressHandler: (handler) =>
      (_sessionStorage.updateProgressHandler = handler),
    getUpdateProgressHandler: () => _sessionStorage.updateProgressHandler,
    setStopProgressHandler: (handler) =>
      (_sessionStorage.stopProgressHandler = handler),
    getStopProgressHandler: () => _sessionStorage.stopProgressHandler,
  },
};
