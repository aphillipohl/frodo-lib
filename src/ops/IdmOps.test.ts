import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { Idm, state } from '../index';
import * as global from '../storage/StaticStorage';
import { mockListAllConfigEntities } from '../test/mocks/ForgeRockApiMockEngine';

const mock = new MockAdapter(axios);

state.default.session.setTenant('https://openam-frodo-dev.forgeblocks.com/am');
state.default.session.setRealm('alpha');
state.default.session.setCookieName('cookieName');
state.default.session.setCookieValue('cookieValue');
state.default.session.setDeploymentType(global.CLOUD_DEPLOYMENT_TYPE_KEY);

describe('IdmOps - countManagedObjects()', () => {
  test('countManagedObjects() 0: Method is implemented', async () => {
    expect(Idm.countManagedObjects).toBeDefined();
  });
});

describe('IdmOps - exportAllConfigEntities()', () => {
  test('exportAllConfigEntities() 0: Method is implemented', async () => {
    expect(Idm.exportAllConfigEntities).toBeDefined();
  });
});

describe('IdmOps - exportAllRawConfigEntities()', () => {
  test('exportAllRawConfigEntities() 0: Method is implemented', async () => {
    expect(Idm.exportAllRawConfigEntities).toBeDefined();
  });
});

describe('IdmOps - exportConfigEntity()', () => {
  test('exportConfigEntity() 0: Method is implemented', async () => {
    expect(Idm.exportConfigEntity).toBeDefined();
  });
});

describe('IdmOps - listAllConfigEntities()', () => {
  test('listAllConfigEntities() 0: Method is implemented', async () => {
    expect(Idm.listAllConfigEntities).toBeDefined();
  });

  test('listSocialProviders() 1: List social identity providers', async () => {
    mockListAllConfigEntities(mock);
    expect.assertions(1);
    await Idm.listAllConfigEntities();
    expect(true).toBeTruthy();
  });
});
