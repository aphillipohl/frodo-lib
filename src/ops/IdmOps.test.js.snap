// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdmOps - importConfigEntity() importConfigEntity() 1: Import a config entity 1`] = `
{
  "_id": "emailTemplate/welcome",
  "defaultLocale": "en",
  "displayName": "Welcome",
  "enabled": true,
  "from": "saas@forgerock.com",
  "html": {
    "en": "<div class="content"><p>Welcome. Your username is '{{object.userName}}'.</p></div>",
  },
  "message": {
    "en": "<html><head></head><body style="background-color: #324054; color: #5e6d82; padding: 60px; text-align: center;"><div class="content" style="background-color: #fff; border-radius: 4px; margin: 0 auto; padding: 48px; width: 235px;"><p>Welcome. Your username is '{{object.userName}}'.</p></div></body></html>",
  },
  "mimeType": "text/html",
  "styles": "body{
 background-color:#324054;
 color:#5e6d82;
 padding:60px;
 text-align:center
}
a{
 text-decoration:none;
 color:#109cf1
}
.content{
 background-color:#fff;
 border-radius:4px;
 margin:0 auto;
 padding:48px;
 width:235px
}
",
  "subject": {
    "en": "Your account has been created",
  },
}
`;
